src/backend/app/Policies/UserPolicy.php

```diff
@@ -18,15 +18,16 @@ public function create(User $currentUser): bool {
     }
 
     public function update(User $currentUser, User $targetUser): bool {
-        // Users can update themselves but not change their own role
-        // roles can be changed by the owner or admin
-        if ($currentUser->id === $targetUser->id) {
-            return true;
-        }
-
         // Check if target user is an owner
         if ($targetUser->hasRole('owner')) {
             // Only users with 'users.manage-owner' permission can update owners
+            // ensure at least one owner exists
+            if (User::whereHas('roles', function ($query) {
+                $query->where('name', 'owner');
+            })->count() === 1) {
+                return false;
+            }
+
```

Comment:
Somehow Github swallowed my comment... Here it is again:

Currently, Owner users cannot modified their own roles at all when they are the only Owner user.

This scenario might occur, when an owner wants to give themselves the admin role in addition to owner to prepare the transfer of owner to someone else. In the current way, they would have to first make someone else an owner, and then remove their own owner role at the same time as giving another role

Reply:
But why not just create an Admin account login and set it up directly before giving them access? Why do they need to "be" and admin first?